client/home/lib/myteam/components/hometeamsendinvites/uploadcsvmodal.coffee

```diff
@@ -0,0 +1,266 @@
+_  = require 'lodash'
+$ = require 'jquery'
+kd = require 'kd'
+ContentModal = require 'app/components/contentModal'
+
+
+module.exports = class UploadCSVModal extends ContentModal
+
+  constructor: (options = {}, data) ->
+
+    { input: @input } = options
+
+    @trailerContent = '''
+      <p>Before uploading your file, please make sure you have the right format.
+      The file should contain <strong>Email</strong>, <strong>First Name</strong> (optional), <strong>Last Name</strong> (optional), and <strong>Role </strong> fields in the right order.
+      Roles can be <strong>Admin</strong> or <strong>Member</strong>.
+      </br></br>
+      <strong>Here is an example:</strong></p>
+      </br>
+      <div class='example-csv'>
+        micheal@example.com, Micheal R., Crawley, Member </br>
+        clora@example.com, Clora J., Ochoa, Member </br>
+        randy@example.com, Randy S., Engel, Admin
+      </div>
+    '''
+
+    options = _.assign {}, options,
+      cssClass: 'content-modal csv-upload'
+      width: 600
+
+    super options
+
+    @setTitle 'Upload CSV File'
+
+    @createTrailerStatePage()
+    @createUploadingStatePage()
+    @createErrorUploadingPage()
+    @createSuccessPage()
+    @creatButtons()
+
+
+  createTrailerStatePage: ->
+
+    @addSubView @trailerPage = new kd.CustomHTMLView
+      tagName: 'main'
+      cssClass: 'main-container trailer-state'
+      partial: @trailerContent
+
+
+  createUploadingStatePage: ->
+
+    @addSubView @uploadingPage = new kd.CustomHTMLView
+      tagName: 'main'
+      cssClass: 'main-container uploading-state hidden'
+
+
+  createErrorUploadingPage: ->
+
+    @addSubView @errorUploading = new kd.CustomHTMLView
+      tagName: 'main'
+      cssClass: 'main-container error-uploading hidden'
+
+
+  createSuccessPage: ->
+
+    @addSubView @successPage = new kd.CustomHTMLView
+      tagName: 'main'
+      cssClass: 'main-container success-state hidden'
+
+
+  creatButtons: ->
+
+    @addSubView @buttonWrapper = new kd.CustomHTMLView
+      cssClass: 'button-wrapper'
+
+    @buttonWrapper.addSubView @selectAndUpload = new kd.ButtonView
+      cssClass: 'GenericButton select-upload'
+      title: 'SELECT AND UPLOAD'
+      callback: =>
+        $('.uploadcsv').change (event) =>
+          @createFormDataAndRequest event
```

Comment:
instead of binding and unbinding the event in global selector scope each time, would this work:

`<input ref={(input) => @input = input} accept='.csv' className='kdinput file uploadcsv' type='file' onBlur={@bound 'createFormDataAndRequest'} />`

or maybe `onChange`?